src/View/Helper/ListsHelper.php

```diff
@@ -309,14 +309,44 @@ public function displayTranslationsDropdown($listId, $translationsLang = null) {
                 'name' => 'translationLangChoice',
                 'languages' => $this->Languages->languagesArrayShowTranslationsIn(),
                 'initialSelection' => $translationsLang,
-                'onSelectedLanguageChange' => "window.location.pathname = '$path' + language.code",
+                'onSelectedLanguageChange' => "window.location.href = '$path' + '$filterLanguage' + '/'+language.code",
```

Comment:
Note that by writing to `href`, you're overwriting the *entire* URL. Using `pathname` here is a deliberate choice, see commit 6781556405b8d30e778890fc0481475b3975d7ab.